feat: add existing components to unit [FC-0083]

rpenido opened the pull request:
## Description

This PR allows adding existing components to units

![image](https://github.com/user-attachments/assets/2126cc23-762f-46b0-a6ef-f87760d356ac)

## Additional Information
- Implements https://github.com/openedx/frontend-app-authoring/issues/1620
- Depends on https://github.com/openedx/frontend-app-authoring/pull/1784

## Testing instructions

- Go to the library home of a library.
- Create a Unit.
- Open the Unit page of the new unit.
- Click on Add content button.
- Click on "Existing Library Content"
- Select some components and click on "Add to Unit"
-  Verify that the blocks are added to the unit

Note: Before merging https://github.com/openedx/frontend-app-authoring/pull/1797 you may see another Units on the component picker
___
Private ref: [FAL-4059](https://tasks.opencraft.com/browse/FAL-4059)

openedx-webhooks commented:
Thanks for the pull request, @rpenido!


This repository is currently maintained by `@openedx/2u-tnl`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

DanielVZ96 reviewed (commented):
besides the same unit appearing in the selection modal, i saw no other issues. lgtm! let me know when that other PR is merged

pomegranited commented:
@rpenido would you be able to address Sam's comment on @ChrisChV 's issue here?
https://github.com/openedx/frontend-app-authoring/issues/1619#issuecomment-2803427297

rpenido commented:
> @rpenido would you be able to address Sam's comment on @ChrisChV 's issue here? [#1619 (comment)](https://github.com/openedx/frontend-app-authoring/issues/1619#issuecomment-2803427297)

Sure @pomegranited! Fixed here: https://github.com/openedx/frontend-app-authoring/pull/1811/commits/504cd32b3a385787f96cdf9471ff9b87886f8ba9

I also fixed a bug that the "Open" button on the Unit Sidebar didn't work inside a collection.
